x-pack/plugins/spaces/server/spaces_service/spaces_service.ts

```diff
@@ -8,15 +8,15 @@ import { map, take } from 'rxjs/operators';
 import { Observable, Subscription, combineLatest } from 'rxjs';
 import { Legacy } from 'kibana';
 import { Logger, KibanaRequest, CoreSetup } from 'src/core/server';
-import { OptionalPlugin } from '../../../../../server/lib/optional_plugin';
-import { DEFAULT_SPACE_ID } from '../../../common/constants';
-import { SecurityPlugin } from '../../../../security';
-import { SpacesClient } from '../../lib/spaces_client';
-import { SpacesConfigType } from '../config';
-import { namespaceToSpaceId, spaceIdToNamespace } from '../../lib/utils/namespace';
+import { SecurityPlugin } from '../../../../legacy/plugins/security';
+import { OptionalPlugin } from '../../../../legacy/server/lib/optional_plugin';
 import { LegacyAPI } from '../plugin';
-import { Space } from '../../../common/model/space';
-import { getSpaceIdFromPath, addSpaceIdToPath } from '../../../common';
+import { SpacesClient } from '../lib/spaces_client';
+import { ConfigType } from '../config';
+import { getSpaceIdFromPath, addSpaceIdToPath } from '../../common/lib/spaces_url_parser';
+import { DEFAULT_SPACE_ID } from '../../common/constants';
+import { spaceIdToNamespace, namespaceToSpaceId } from '../lib/utils/namespace';
+import { Space } from '../../common/model/space';
 
 type RequestFacade = KibanaRequest | Legacy.Request;
```

Comment:
I think that's a great idea!

Reply:
Actually I'm going to revert this change - the spaces service still requires the `getBasePath` function that exists on the LP request and the "fake" requests used by alerting and reporting. I'd rather come up with a holistic solution for these fake requests in a dedicated PR since it will likely end up touching these other plugins as well

Reply:
Ugh, I see :sob:  Tackling this in separately sounds good to me.